deepchem/models/torch_models/cnn.py

```diff
@@ -0,0 +1,442 @@
+    elif pool_type == 'max':
+      PoolLayer = (F.max_pool1d, F.max_pool2d, F.max_pool3d)[self.dims - 1]
+    else:
+      raise ValueError("pool_type must be either 'average' or 'max'")
+
+    self.PoolLayer = PoolLayer
+    self.layers = nn.ModuleList()
+
+    in_shape = n_features
+
+    for out_shape, size, stride, dropout, weight_stddev, bias_const in zip(
+        layer_filters, kernel_size, strides, dropouts, weight_init_stddevs,
+        bias_init_consts):
+
+      block = nn.Sequential()
+
+      layer = ConvLayer(in_channels=in_shape,
+                        out_channels=out_shape,
+                        kernel_size=size,
+                        stride=stride,
+                        padding=padding,
+                        dilation=1,
+                        groups=1,
+                        bias=True)
+
+      nn.init.normal_(layer.weight, 0, weight_stddev)
+
+      # initializing layer bias with nn.init gives mypy typecheck error
+      # using the following workaround
+      if layer.bias is not None:
+        layer.bias = nn.Parameter(torch.full(layer.bias.shape, bias_const))
+
+      block.append(layer)
+
+      block.append(nn.Dropout(dropout))
+
+      self.layers.append(block)
+
+      in_shape = out_shape
+
+    self.classifier_ffn = nn.LazyLinear(self.n_tasks * self.n_classes)
+    self.regressor_ffn1 = nn.LazyLinear(self.n_tasks)
+    self.regressor_ffn2 = nn.LazyLinear(self.n_tasks)
+
+  def forward(self, x: torch.Tensor) -> List[Any]:
+    """
+    Parameters
+    ----------
+    x: torch.Tensor
+      Input Tensor
+
+    Returns
+    -------
+    torch.Tensor
+      Output as per use case : regression/classification
+    """
+    x = torch.transpose(x, 1, -1)  # n h w c -> n c h w
+
+    prev_layer = x
+
+    for layer, activation_fn in zip(self.layers, self.activation_fns):
+      x = layer(x)
+      # residual blocks can only be used when successive layers have the same output shape
+      if self.residual and x.shape[1] == prev_layer.shape[1]:
+        x = x + prev_layer
+
+      if activation_fn is not None:
+        x = activation_fn(x)
+
+      prev_layer = x
+
+    x = self.PoolLayer(x, kernel_size=x.size()[2:])
+
+    outputs = []
+    batch_size = x.shape[0]
+
+    x = torch.reshape(x, (batch_size, -1))
+
+    if self.mode == "classification":
+
+      logits = self.classifier_ffn(x)
+      logits = logits.view(batch_size, self.n_tasks, self.n_classes)
+      output = F.softmax(logits, dim=2)
+      outputs = [output, logits]
+
+    else:
+      output = self.regressor_ffn1(x)
+      output = output.view(batch_size, self.n_tasks)
+
+      if self.uncertainty:
+        log_var = self.regressor_ffn2(x)
+        log_var = log_var.view(batch_size, self.n_tasks, 1)
+        var = torch.exp(log_var)
+        outputs = [output, var, output, log_var]
+
+      else:
+        outputs = [output]
+
+    return outputs
+
+
+class CNN(TorchModel):
```

Comment:
Copy over the old docstring from `CNN` here and make sure to add a usage example